File: leetcode/js/longest-continuous-increasing-subsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findLengthOfLCIS = function (nums) {
  let result = 1;
  let count = 1;

  for (let i = 0; i < nums.length - 1; i++) {
    if (nums[i] < nums[i + 1]) {
      count++;
    } else {
      result = Math.max(result, count);
      count = 1;
    }
  }

  return Math.max(result, count);
};

File: leetcode/rust/longest-continuous-increasing-subsequence.rs
impl Solution {
    pub fn find_length_of_lcis(nums: Vec<i32>) -> i32 {
        let mut result = 1;
        let mut count = 1;

        for i in 0..nums.len() - 1 {
            if nums[i] < nums[i + 1] {
                count += 1;
            } else {
                result = result.max(count);
                count = 1;
            }
        }

        result.max(count)
    }
}
